File: 3/keypad.rb
class KeypadKata

  def initialize(dictionary_path)
    @words = File.readlines(dictionary_path).map { |word| word.strip }
  end



  def predict(input)
    map_numbers(input.split(' ')).join
  end



  private

  def map_numbers(numbers)
    coding_table = {
      '2' => 'abc',
      '3' => 'def',
      '4' => 'ghi',
      '5' => 'jkl',
      '6' => 'mno',
      '7' => 'pqrs',
      '8' => 'tuv',
      '9' => 'wxyz'}

    numbers.map do |number|
      coding_table[number[0]][number.length-1]
    end
  end


end

File: 4/Keypad.php
<?php

class Keypad
{

    private $keyMap = array(
        '2' => 'ABC',
        '3' => 'DEF',
        '4' => 'GHI',
        '5' => 'JKL',
        '6' => 'MNO',
        '7' => 'PQRS',
        '8' => 'TUV',
        '9' => 'WXYZ'
    );



    public function getWords($input)
    {
        $words = array();
        if (strlen($input))
        {
            $letterDefinitions = explode(' ', $input);
            $word = '';
            foreach ($letterDefinitions as $letter)
            {
                $word .= $this->getLetterFromNumberSequence($letter);
            }
            $words[] = $word;
        }
        return $words;
    }



    private function getLetterFromNumberSequence($input)
    {
        $baseLetters = $this->keyMap[$input[0]];
        return $baseLetters[strlen($input) - 1];
    }

}